pkg/costmodel/allocation.go

```diff
@@ -1383,16 +1382,15 @@ func applyNodeCostPerCPUHr(nodeMap map[nodeKey]*NodePricing, resNodeCostPerCPUHr
 			nodeMap[key] = &NodePricing{
 				Name:       node,
 				NodeType:   instanceType,
-				ProviderID: providerIDParser(providerID),
+				ProviderID: cloud.ParseID(providerID),
```

Comment:
Is there something we could do so that we don't have to parse the `providerID` at this point? Feels like a step that we could (maybe even should) be doing upstream of this.

Reply:
The step before this is metric emission into Prometheus, I really like the idea of all the providerIDs going into Prometheus being pre parsed. Two things about that though. There would be a coverage gap, where we would have to parse it on both sides until we were sure that all the unparsed providers ids were out of everyones prometheus. I know we also don't emit all of the metrics that we use, if there were some that has provider_id and we were not in control we would still have to parse them after which would end up being a mixed solution with some pre-parsing and some post-parsing, which I would rather not do. Off hand do you know if that is the case? If we were able to do pre parsing I could move the parse functions back into the provider implementations which would be nice and allow for meaningful logging on parsing too.

Reply:
`resNodeCostPerCPUHr` could theoretically contain results for multiple cloud providers though, right?

Reply:
Ah, looks like I was a little slow. That makes perfect sense Sean.